internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -718,6 +718,24 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 							ValidateFunc: validate.CIDR,
 						},
 
+						"ebpf_data_plane": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(managedclusters.EbpfDataplaneCilium),
+							}, false),
+						},
+
+						"network_plugin_mode": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"overlay",
```

Comment:
There's an enum for this `managedclusters.NetworkPluginModeOverlay` which we should be using. I see that the API returns this as lowercase so can you please 
* raise a bug on the API spec repo that the casing isn't being maintained
* normalise the value in the read function
* leave a comment above the normalisation in the read function referencing the bug raised on the API spec repo